presto-hive/src/main/java/io/prestosql/plugin/hive/util/HiveWriteUtils.java

```diff
@@ -556,9 +565,43 @@ public static Path createTemporaryPath(ConnectorSession session, HdfsContext con
 
         createDirectory(context, hdfsEnvironment, temporaryPath);
 
+        if (hdfsImpersonationEnabled) {
```

Comment:
@findepi According to your advice, I put the operation here to apply the permissions right after creating the file in tmp